common/persistence/sql/common.go

```diff
@@ -35,8 +37,10 @@ import (
 
 // TODO: Rename all SQL Managers to Stores
 type sqlStore struct {
-	db     sqlplugin.DB
-	logger log.Logger
+	db      sqlplugin.DB
+	logger  log.Logger
+	encoder serialization.Encoder
```

Comment:
Related to my comment above: Is there any reason for keeping encoder and decoder interfaces separate? I think this separation bloats the code a little bit and leaves the door open for using incompatible encoder & decoder pairs in the future. It seems like you can just have one _parser_ that handles both encoding and decoding.